### PR TITLE
Implement /api/editusertext in SubmitActions

### PR DESCRIPTION
**Pull request by Anseriform**
Implemented the editusertext api call as editUserText(String fullname, String text) in SubmitActions. 
This handles editing of comments and selfposts.
No test class though...

### CODE REVIEW

**Comment by sfat** [comment]
hey @Anseriform, thanks for the contribution! Did you had some difficulties in regards to writing some unit tests? You need help or something?

File: src/main/java/com/github/jreddit/action/SubmitActions.java [hunk 1]
```diff
@@ -216,4 +216,40 @@ else if (object.toJSONString().contains(".error.BAD_CAPTCHA.field-captcha")) {
...
+     */
+    public boolean editUserText(String fullname, String text) throws ActionFailedException {
+
+        JSONObject object = (JSONObject) restClient.post(
+                "thing_id=" + fullname + "&text=" + text + "&uh=" + user.getModhash(),
+                ApiEndpointUtils.EDITUSERTEXT,
+                user.getCookie()
+        ).getResponseObject();
+
+        if (object.toJSONString().contains(".error.USER_REQUIRED")) {
```
Comment (sfat) [inline src/main/java/com/github/jreddit/action/SubmitActions.java]: you can make a local variable to keep the JSON String to not call too many times ```object.toJSONString()```. something like this:
```
String responseAsString = object.toJSONString();
````
then you can just make use of the variable declared.

Also for the returns if it was successful or not, just make a boolean variable and just set that variable if it's okay or not.
Example:
```
boolean successful = true;

if (responseAsTring.contains(".error.NOT_AUTHOR")) {
   successful = false;
}

....

return successful;
```

Not sure about the chained if's. I think we have in other places in the code where we make use of switch case. 
I see that you are making use of ```.contains()```. Does the response contain something else than just ".error.NOT_AUTHOR", for instance?

File: src/main/java/com/github/jreddit/action/SubmitActions.java [hunk 2]
```diff
@@ -216,4 +216,40 @@ else if (object.toJSONString().contains(".error.BAD_CAPTCHA.field-captcha")) {
...
+    /**
+     * This function edits the text of a comment or submission (self). Requires
+     * authentication.
+     *
+     * @param text The new text for the comment or selfpost
+     * @param fullname The fullname of the submission or comment
+     * @return Whether the edit succeeded
+     * @throws ActionFailedException If the action failed
+     */
+    public boolean editUserText(String fullname, String text) throws ActionFailedException {
```
Comment (sfat) [inline src/main/java/com/github/jreddit/action/SubmitActions.java]: what happens if one of the parameters are null?

**Comment by Anseriform** [comment]
@sfat thanks for the feedback.

I have never worked with jUnit tests before and the TestSubmitActions.java class has no code, so I had no example to start working from.

About the chained if's:
this is what the response looks like if you try to edit a comment that's not yours:
```
{"jquery":[[0,1,"call",["body"]],[1,2,"attr","find"],[2,3,"call",[".status"]],[3,4,"attr","hide"],[4,5,"call",[]],[5,6,"attr","html"],[6,7,"call",[""]],[7,8,"attr","end"],[8,9,"call",[]],[1,10,"attr","find"],[10,11,"call",[".error.NOT_AUTHOR.field-thing_id"]],[11,12,"attr","show"],[12,13,"call",[]],[13,14,"attr","text"],[14,15,"call",["you can't do that"]],[15,16,"attr","end"],[16,17,"call",[]]]}
```

I guess it's possible to get the error message out of the JSONArray with a get() and use it in a switch-case statement, but personally I think using `.toString()` and `.contains()` would be more reliable. 
To make editUserText() I used the submit function in SubmitActions as example which has the chained if's as well.
I think using `return false` immediately is OK because there is no need to continue checking for other errors once an error is found.

If the fullname parameter is null, reddit will return the NOT_AUTHOR error. 
If text is null, it will edit the submission's/comment's (self)text to the word "null".

**Comment by sfat** [comment]
@Anseriform you can take a look for instance at SubredditsRetrievalTest to see how we do tests. You can try and give it a shot and I can help you out with the details. If you don't manage then let me know and I will merge this PR and add a test myself

**Comment by Anseriform** [comment]
@sfat I looked up some info on jUnit and Mockito and I'm almost done with TestSubmitActions.

Also, I noticed SubmitActions.delete() doesn't work, the constant in ApiEndpointUtils is "/delete" while it should be "/del" according to reddit.com/api, can I fix this?

**Comment by sfat** [comment]
@Anseriform sure. if you wanna to it in this PR, I'll revert the merge and you can follow up in this one.

**Pull request merged**